src/components/elements/FilterChips.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import Chip from 'material-ui/Chip'
+import { removeFilter  } from '../../actions/filterActions'
+
+const FilterChips = ({dispatch, filter, filterType, filterOptions, lookupMethod, history}) => {
+    if (readyToRender(filter, filterOptions)) {
+        let selectedFilters = extractWantedFilter(filterType, filter)
+        return (
+            <div style={chipStyles.wrapper}>
+                {selectedFilters.map((selectedFilter) => renderChip(dispatch, history, filter, selectedFilter, filterOptions, lookupMethod))
+                }
+            </div>
+        )
+    }
+    else {
+        return <div></div>
+    }
+
+}
+
+const renderChip = (dispatch, history, filter, filterOptions, selectedFilterType, lookupMethod) => {
```

Comment:
This method would be more focused if it only took three args: The id, the name, and a function that when called with no arguments dispatches the (already populated) removeFilter action. Separate the concern of actually rendering the chip from the concern of transforming the data.